tests/test_web_response.py

```diff
@@ -1171,3 +1173,36 @@ def test_text_is_json_encoded(self) -> None:
     def test_content_type_is_overrideable(self) -> None:
         resp = json_response({"foo": 42}, content_type="application/vnd.json+api")
         assert "application/vnd.json+api" == resp.content_type
+
+
+@pytest.mark.skipif(
+    not bool(os.environ.get("PYTHONASYNCIODEBUG")),
+    reason="Only run the test with debug mode",
+)
+async def test_warn_large_cookie(buf: Any, writer: Any) -> None:
+    resp = Response()
+    resp.set_cookie("foo", "8" * 4070, max_age=2600)  # No warning
```

Comment:
To ensure we are testing bytes. Let's do something like:
`value = "ÿ" + "8" * 4068`
And same again for the second test below.

Reply:
Probably for reuse and consistency. Anyway, I don't think there should be more than one test in a test.

Reply:
I am not sure why `"ÿ"` takes 6 bytes when calculating the length of the cookie header but `len('ÿ'.encode("utf-8"))` gives 2, maybe because `len('ÿ'.encode("utf-8"))` returns a byte string.

Reply:
It's not encoded to utf-8:
```
>>> from http.cookies import _Translator
>>> "ÿ".translate(_Translator)
'\\377'
```
It's 4 bytes, but I think it may also result in adding `"` around the cookie value for another 2 bytes.

Reply:
Not sure if Chrome counts it this way, or after translating it back to utf-8 or something. But, following the pessimistic approach, counting this encoded value is the safe option. It is only a warning after all.